src/web/components/BylineLink.tsx

```diff
@@ -9,6 +9,42 @@ const getContributorTags = (tags: TagType[]): TagType[] => {
 	return tags.filter((t) => t.type === 'Contributor');
 };
 
+const applyCleverOrderingForMatching = (titles: string[]): string[] => {
+	/*
+		Q: Why does this function exist ?
+
+		A: We had a case where the byline was "Sam Levin in Los Angeles and Sam Levine in New York",
+		which would lead to the regex: Sam Levin|Sam Levine. Unfortunately the first expression would take priority and we would end up
+		with a bylineAsTokens equal to [
+			'',
+			'Sam Levin',
+			' in Los Angeles and ',
+			'Sam Levin',
+			'e in New York'
+		]
+
+		The solution here is simply to order the titles in decreasing length. This ensures that in a case where one name is a substring or another
+		name, then the longest name is tried first, preventing the problem we had. We now have a bylineAsTokens equals to [
+			'',
+			'Sam Levin',
+			' in Los Angeles and ',
+			'Sam Levine',
+			' in New York'
+		]
+
+		This function doesn't change any thing for the "standard" case, but solves the "Sam Levin|Sam Levine" case and similar cases.
+
+		ps: If one day another problem comes up whose solution would be incompatible with this correction, then a better solution for the matching
+		will have to be invented from the ground up, but for the moment, this will do :)
+	*/
+
+	return titles
+		.sort((a, b) => {
+			return a.localeCompare(b);
+		})
+		.reverse();
+};
+
 // This crazy function aims to split bylines such as
```

Comment:
I think this comment was lifted from Frontend but it seems things are getting crazier!

But seriously, should be centralise the comments/descriptions to this one place?